### DISCUSSION TITLE
Athom 4ch JSON

### DISCUSSION THREAD

**Discussion by rtebra** [RavenSystem/esp-homekit-devices#1824] [MEPLHAA Scripts Collection]
Has anyone successfully flashed a Athom 4ch with a suitable JSON script? Flashing was successfully on my first try but I used the Sonoff 4ch JSON and that doesn’t work. It works with relay 3 but not for the others.

**Comment by andyallcorn** [comment]
You need to change the GPIOs that the relays work on to match your hardware.

A quick Google has brought up this information about the hardware, hopefully this is what you have: https://templates.blakadder.com/athom_R04-30A.html

Now, look at your template JSON, and change the GPIO numbers that get changed in the “0” and “1” sections of each accessory to match the numbers you can see I. That page. That’s the values next to “g” inside the “r” section of your JSON. There will be 4 pairs of “g” values you need to change, one pair to each of the relay GPIOs.

**Reply by rtebra** [reply]
Thank for your reply. I get a warning for an invalid JSON. This is what I used:
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":14}]},"1":{"r":[{"g":14,"v":1}]},"b":[{"g":14}]},{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]},{“0”:{“r”:[{“g”:13}]},“1”:{“r”:[{“g”:13,”v”:1}]},“b”:[{“g”:9}]},{“0”:{“r”:[{“g”:16}]},“1”:{“r”:[{“g”:16,”v”:1}]},"b":[{"g":10}]}]} 
Can you help me out?

**Reply by pawelosin** [reply]
All quotation marks must be of this type "

**Reply by rtebra** [reply]
The quotation marks validated the JSON string.

**Reply by VaBanck** [reply]
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":14}]},"1":{"r":[{"g":14,"v":1}]},"b":[{"g":14}]},{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]},{"0":{"r":[{"g":13}]},"1":{"r":[{"g":13,"v":1}]},"b":[{"g":9}]},{"0":{"r":[{"g":16}]},"1":{"r":[{"g":16,"v":1}]},"b":[{"g":10}]}]}

**Reply by litris** [reply]
And in push button mode 0.5s how would it be?

I got it with switch type but in HomeKit visually a switch appears.

I scrolled to ON and it activates the relay but the button visually does not go back to off I have to turn it down to Off.

I use the 8 Relay board with this code taken from HAA configured.

And I don't get a button to press only a switch appears I don't know if it's a bug.

Thank you.

 {"c":{"o":0},"a":[{"0":{"r":[{"g":16,"v":0}]},"1":{"r":[{"g":16,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":14,"v":0}]},"1":{"r":[{"g":14,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":13,"v":0}]},"1":{"r":[{"g":13,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":15,"v":0}]},"1":{"r":[{"g":15,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":0,"v":0}]},"1":{"r":[{"g":0,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":4,"v":0}]},"1":{"r":[{"g":4,"v":1,"i":0.5}]},"t":1,"b":[],"s":0},{"0":{"r":[{"g":5,"v":0}]},"1":{"r":[{"g":5,"v":1,"i":0.5}]},"t":1,"b":[],"s":0}]}
